lib/src/rules/prefer_final_fields.dart

```diff
@@ -64,6 +63,25 @@ class GoodMutable {
 ```
 ''';
 
+bool _isMutated(
+        VariableDeclaration variable, CompilationUnit compilationUnit) =>
+    DartTypeUtilities
+        .traverseNodesInDFS(compilationUnit)
+        .where((n) =>
```

Comment:
Would `any` allow us to short circuit the traversal as soon as we found one matching node?